src/coreComponents/constitutive/solid/SolidModelDiscretizationOps.hpp

```diff
@@ -0,0 +1,134 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2019 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2019 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2019 Total, S.A
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All right reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file SolidModelHelperBase.hpp
+ */
+
+
+#ifndef GEOSX_CONSTITUTIVE_SOLID_SOLIDMODELDISCRETIZATIONOPS_HPP_
+#define GEOSX_CONSTITUTIVE_SOLID_SOLIDMODELDISCRETIZATIONOPS_HPP_
+
+#include "common/DataTypes.hpp"
+
+namespace geosx
+{
+namespace constitutive
+{
+
+
+struct SolidModelDiscretizationOps
```

Comment:
I would rename `SolidModelDiscretizationOps` (and derived specializations) to be more specific. Functions `BTDB`, `diagBTDB` and `diagRowSumBTDB` are relative to the Galerkin (i.e. `BT = transpose(B)`)  finite element approach.